website/docs/r/cognitive_account_customer_managed_key.html.markdown

```diff
@@ -0,0 +1,118 @@
+---
+subcategory: "Cognitive Services"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cognitive_account_customer_managed_key"
+description: |-
+  Manages a Customer Managed Key for a Cognitive Services Account.
+---
+
+# azurerm_cognitive_account_customer_managed_key
+
+Manages a Customer Managed Key for a Cognitive Services Account.
+
+## Example Usage
+
+```hcl
+data "azurerm_client_config" "current" {}
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West US"
+}
+resource "azurerm_user_assigned_identity" "example" {
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  name                = "example-identity"
+}
+resource "azurerm_cognitive_account" "example" {
+  name                  = "example-account"
+  location              = azurerm_resource_group.example.location
+  resource_group_name   = azurerm_resource_group.example.name
+  kind                  = "Face"
+  sku_name              = "E0"
+  custom_subdomain_name = "example-account"
+  identity {
+    type         = "SystemAssigned, UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.example.id]
+  }
+}
+resource "azurerm_key_vault" "example" {
+  name                     = "example-vault"
+  location                 = azurerm_resource_group.example.location
+  resource_group_name      = azurerm_resource_group.example.name
+  tenant_id                = data.azurerm_client_config.current.tenant_id
+  sku_name                 = "standard"
+  soft_delete_enabled      = true
+  purge_protection_enabled = true
+}
+resource "azurerm_key_vault_access_policy" "cognitive" {
+  key_vault_id       = azurerm_key_vault.example.id
+  tenant_id          = azurerm_cognitive_account.example.identity.0.tenant_id
+  object_id          = azurerm_cognitive_account.example.identity.0.principal_id
+  key_permissions    = ["get", "create", "list", "restore", "recover", "unwrapkey", "wrapkey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
+resource "azurerm_key_vault_access_policy" "client" {
+  key_vault_id       = azurerm_key_vault.example.id
+  tenant_id          = data.azurerm_client_config.current.tenant_id
+  object_id          = data.azurerm_client_config.current.object_id
+  key_permissions    = ["get", "create", "delete", "list", "restore", "recover", "unwrapkey", "wrapkey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
+resource "azurerm_key_vault_access_policy" "user" {
+  key_vault_id       = azurerm_key_vault.example.id
+  tenant_id          = azurerm_user_assigned_identity.example.tenant_id
+  object_id          = azurerm_user_assigned_identity.example.principal_id
+  key_permissions    = ["get", "create", "delete", "list", "restore", "recover", "unwrapkey", "wrapkey", "purge", "encrypt", "decrypt", "sign", "verify"]
+  secret_permissions = ["get"]
+}
```

Comment:
same here, if we inline these then we can remove these 3 resources & the depends_on